src/System.IO.Compression/src/Interop/Interop.zlib.cs

```diff
@@ -49,6 +52,14 @@ internal static partial class zlib
             }
         }
 
+        internal static unsafe uint crc32(uint crc, byte[] buffer, int offset, int len)
+        {
+            if (buffer == null)
+                return crc32(crc, null, len);
```

Comment:
This now means we have an extra branch on every crc32 call.  While it's probably not measurable, it doesn't seem worthwhile.  Why can't the caller just do this directly if it knows it has a null buffer?